### ISSUE TITLE
Taking pictures with SONY RX0 M2 doesn't work

### ISSUE THREAD

**Issue by PLMR-FR** [gphoto/gphoto2#349]
**Camera model :** 

Sony DSC RX0 II (PC Control)   usb:001,011

**gphoto2 version :**

$ gphoto2 --version
gphoto2 2.5.23

Copyright (c) 2000-2019 Marcus Meissner and others

gphoto2 comes with NO WARRANTY, to the extent permitted by law. You may
redistribute copies of gphoto2 under the terms of the GNU General Public
License. For more information about these matters, see the files named COPYING.

This version of gphoto2 is using the following software versions and options:
gphoto2         2.5.23         gcc, popt(m), exif, no cdk, no aa, jpeg, no readline
libgphoto2      2.5.25         standard camlibs, gcc, ltdl, EXIF
libgphoto2_port 0.12.0         iolibs: disk ptpip serial usb1 usbdiskdirect usbscsi, gcc, ltdl, EXIF, USB, serial without locking

**To reproduce:**

Connect the camera and execute : $ gphoto2 --capture-image
When doing this I can hear "Bip" sound from autofocus but it doesn't take the picture in AF mode and it leaves the instruction after 40 seconds with this output:
```
ERROR: Could not capture image.
ERROR: Could not capture.
```

When I am in MF mode, sometimes it takes the picture but I am still blocked for 40 seconds with the same output:
```
ERROR: Could not capture image.
ERROR: Could not capture.
```

Here is the output of gphoto2 --capture-image --debug --debug-logfile gphoto_debug_take_picture_failed.txt

[gphoto_debug_take_picture_failed.txt.zip](https://github.com/gphoto/gphoto2/files/4908914/gphoto_debug_take_picture_failed.txt.zip)

Thanks for your help.

**Comment by msmeissn** [comment]
can you try
gphoto2 --wait-event=3s --capture-image-and-download

and see if that helps?

**Comment by PLMR-FR** [comment]
Hello,

Yes it's better, now I am able to take pictures with autofocus. But it is still unstable sometimes it still doesn't work. I would say that it works 75% of the time.

Here is the output when it works:

```
gphoto2 --wait-event=3s --capture-image-and-download
Waiting for 3 seconds for events from camera. Press Ctrl-C to abort.           
UNKNOWN PTP Property 0000 changed
UNKNOWN PTP Property 0000 changed
UNKNOWN PTP Property 0000 changed
New file is in location /capt0000.jpg on the camera
```

Here is the output when it doesn't work

```
gphoto2 --wait-event=3s --capture-image-and-download
Waiting for 3 seconds for events from camera. Press Ctrl-C to abort.           
UNKNOWN PTP Property 0000 changed
UNKNOWN PTP Property 0000 changed
UNKNOWN PTP Property 0000 changed
ERROR: Could not capture image.
ERROR: Could not capture.
```
Is a debug log would be useful ?

Any chance to not have to wait 3 seconds for taking a picture ?

Thanks for your support !

**Comment by msmeissn** [comment]
i encountered similar issues with the Sony ZV-1 just 2 weeks ago. :/
Perhaps make it 4 seconds or more and see if it gets more stable?

For the ZV-1 I looked for indicators that the camera would be ready, but I did not find any at that time.

**Comment by PLMR-FR** [comment]
Thanks I tried with 4 seconds and now it seems to works every time.

One other thing, I want to use the Burst mode. But when I set this mode it still takes only one picture. Is there any way to have a "start" and "stop" command for Burst mode ?

Thank you very much for your support

**Comment by msmeissn** [comment]
I am not aware of how burst mode works on sony and if it can be configured from remote ... the parameters we have do not seem to show it. hmm

**Comment by TimRyan99** [comment]
I have the same issues with the Sony RX0 M2 using gphoto2 --capture-image-and-download. The camera hangs until I get this error message after a long wait - perhaps 30-40 seconds. 
ERROR: Could not capture image.
ERROR: Could not capture.
I have been working on Linux Mint 20 running on a USB drive. I have tested on Dell latitude and small Asus laptop. Same outcome. Have also tried using latest Fedora install with no success. 
I tested with above suggestion to use gphoto2 --wait-event=3s --capture-image-and-download and seems to be reliable. If I shorten this to 1s it hangs just like when no wait event is used. 

I need to be taking photos as quickly as possible and a 3 second delay is a show stopper for what I am trying to achieve

Attached are three debug files. SonyRX0-wait-event3s.txt (this shows what success looks like), SonyRX0---wait-event1s.txt (this hangs) and capture_and_download.txt. 
I have camera set to manual with no autofocus. 

[SonyRX0-wait-event3s.txt](https://github.com/gphoto/gphoto2/files/5146110/SonyRX0-wait-event3s.txt)
[SonyRX0-wait-event1s.txt](https://github.com/gphoto/gphoto2/files/5146112/SonyRX0-wait-event1s.txt)
[capture_and_download.txt](https://github.com/gphoto/gphoto2/files/5146113/capture_and_download.txt)

Appreciate any assistance on this. Thank you

**Comment by TimRyan99** [comment]
Just to add, camera was set to manual focus. In this mode gphoto2 --trigger-capture works fine. If I set it to autofocus and run gphoto2 --trigger-capture then camera goes through focus stage but does not trigger. gphoto2 --wait-event=3s --capture-image-and-download seems to work the same whether in manual or autofocus.

Then I set to autofocus and ran this: gphoto2 --trigger-capture --capture-image-and-download and found that lo and behold the camera took a photo and downloaded to the pc. It seemed a little slow but that might be the time it was taking for the focus to be found. This might be a workaround that looks to react much faster than using the wait-event. Note however this method does not work if the camera is set to manual focus. It would be nice to see gphoto2 --capture-image-and-download work for manual mode without having to put an arbitary delay in place. 

Hope this workaround is of use to people

**Comment by TimRyan99** [comment]
Hi there, wondering if there has been any movement on this issue? I have been using python gphoto2 with a capture command in a loop. 
captureCommand = ["--wait-event=3s --capture-image-and-download" ,'myfilename.jpg'] 
gp(captureCommand)
this does not trigger a camera image and download whereas if I manually do this same from the command line it works fine.

My preference is to simply have the command without the wait event.
captureCommand = ["--capture-image-and-download" ,'myfilename.jpg'] 
This does trigger a camera event and images is downloaded, but the original issue of needing a super long wait (30-40s) before the camera releases to allow another photo to be taken remains. Appreciate any feedback.

**Comment by msmeissn** [comment]
thanks for the datapoint regarding the in "manual focus" mode that this delays seems not required.

I focused on other cameras in the last months, so I did not had access to a recent Sony. I will need to debug this when I can rent one :/

**Comment by gabrielecoppi** [comment]
I am having a very similar issue. The 3s workaround seems fine if I need to take a single image. However, I need to take images every 1s and if I run the interval with that wait, some images are taken, some are skipped (no particular order the skipped one). I also tried to set the manual focusing and add the `--capture-trigger` flag, but I can hear the camera shooting and then I got 'ERROR CAPTURE'  [LOG FILE trigger.zip](https://github.com/gphoto/gphoto2/files/7970549/trigger.zip). I specify that I am on Ubuntu 21.10 and I am connected to the camera using a USB-C adapter
